test/unit/lib/three_scale/bot_protection/form_test.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ThreeScale::BotProtection::FormTest < ActiveSupport::TestCase
+
+  class Test
+    include ThreeScale::BotProtection::Form
+  end
+
+  setup do
+    @account = FactoryBot.create(:buyer_account)
+    @subject = Test.new.tap do |s|
+      template = mock('template')
+      template.stubs(:site_account).returns(@account)
+      s.stubs(:template).returns(template)
+    end
+  end
+
+  class BotProtectionLevel < ThreeScale::BotProtection::FormTest
+    test 'bot_protection_level takes the value from provider settings' do
+      @account.settings.update(spam_protection_level: :captcha)
+
+      result = @subject.send :bot_protection_level
+
+      assert_equal :captcha, result
+    end
+  end
+
+  class BotProtectionEnabled < ThreeScale::BotProtection::FormTest
+    test 'bot_protection_enabled? is false when reCaptcha is not configured' do
+      Recaptcha.stubs(:captcha_configured?).returns(false)
+
+      result = @subject.send :bot_protection_enabled?
+
+      assert_not result
+    end
+
+    test 'bot_protection_enabled? is false when the bot protection is disabled' do
+      @account.settings.update(spam_protection_level: :none)
+
+      result = @subject.send :bot_protection_enabled?
+
+      assert_not result
+    end
+
+    test 'bot_protection_enabled? is false when the user is logged in' do
+      @subject.template.stubs(:logged_in?).returns(true)
+
+      result = @subject.send :bot_protection_enabled?
+
+      assert_not result
+    end
+
+    test 'bot_protection_enabled? is true when reCaptcha is configured and the bot protection is enabled' do
+      Recaptcha.stubs(:captcha_configured?).returns(true)
+      @subject.template.stubs(:logged_in?).returns(false)
+      @account.settings.update(spam_protection_level: :captcha)
+
+      result = @subject.send :bot_protection_enabled?
+
+      assert result
+    end
+  end
+
+  class BotProtectionInputs < ThreeScale::BotProtection::FormTest
+    test 'bot_protection_inputs returns nothing if bot protection is disabled' do
+      @subject.stubs(:bot_protection_enabled?).returns(false)
+
+      result = @subject.send :bot_protection_inputs
+
+      assert_empty result
+    end
+
+    test 'bot_protection_inputs returns reCaptcha if bot protection is disabled' do
+      @subject.stubs(:bot_protection_enabled?).returns(true)
+      @subject.template.stubs(:controller).returns(mock('controller', controller_path: '/test'))
+
+      result = TestOutputBuffer.new
+      result.concat(@subject.send(:bot_protection_inputs))
```

Comment:
could you explain why do we need to use this Buffer class?

Reply:
We don't need it. I took that from another test, but I changed it to be more simple, in https://github.com/3scale/porta/commit/16ae49bbca20572fabc220c1e80450fbd44697c1